### PR TITLE
Fix #32 Convert to async await.

### LINKED ISSUE #32: Convert to async/await
I think we're clear to use those without any transpilation.

### PR DESCRIPTION
**Pull request by fzzzy**

### CODE REVIEW

File: addon/background.js [hunk 1]
```diff
@@ -21,20 +21,20 @@ browser.contextMenus.create({
...
+    await browser.sidebarAction.open();
+    const windowInfo = await browser.windows.getCurrent();
     let desktop = !!desktopHostnames[(new URL(url)).hostname];
     let message = {type: "browse", url, windowId: windowInfo.id, desktop};
     return retry(() => {
       return browser.runtime.sendMessage(message);
     }, {times: 3, wait: 50});
-  }).catch((error) => {
+  } catch (error) {
     console.error("Error setting panel to page:", error);
```
Comment (ianb) [inline addon/background.js]: We might want to consider if these errors are reported fine on their own. It's not great with `.catch()`, but maybe this works fine? Or not... worth experimenting I guess.

File: addon/background.js [hunk 2]
```diff
@@ -105,22 +99,35 @@ chrome.webRequest.onHeadersReceived.addListener(function (info) {
...
 
 function timeout(time) {
   return new Promise((resolve) => {
     setTimeout(resolve, time);
   });
 }
+
+async function init() {
+  try {
+    const result = browser.storage.sync.get("desktopHostnames");
```
Comment (ianb) [inline addon/background.js]: Doesn't this have to be `const result = await browser.storage.sync.get("desktopHostnames");` ?

File: addon/sidebar.js [hunk 3]
```diff
@@ -22,31 +11,33 @@ function displayPage(url, desktop) {
...
-      anchor.prepend(image);
-      li.appendChild(anchor);
-      tabList.appendChild(li);
-    }
-  }).catch((error) => {
+  let windowInfo;
+  try {
+    windowInfo = await browser.windows.getCurrent({populate: true});
+  } catch (error) {
     console.error("Error getting window info:", error);
```
Comment (ianb) [inline addon/sidebar.js]: This translation isn't quite right – failing to get the windowInfo should be fatal (and probably will be a fatal TypeError later). We might not need any try/catch in this function at all.

**Comment by fzzzy** [comment]
I merged the GA pr, so this needs rebasing now.

**Pull request merged**